bindings/go/rsa/signing/handler/handler.go

```diff
@@ -0,0 +1,292 @@
+		return descruntime.SignatureInfo{}, fmt.Errorf("convert config: %w", err)
+	}
+	algorithm := supported.GetSignatureAlgorithm()
+
+	priv := rsacredentials.PrivateKeyFromCredentials(creds)
+	if priv == nil {
+		return descruntime.SignatureInfo{}, ErrMissingPrivateKey
+	}
+
+	hash, dig, err := parseDigest(unsigned)
+	if err != nil {
+		return descruntime.SignatureInfo{}, err
+	}
+
+	rawSig, err := signRSA(algorithm, priv, hash, dig)
+	if err != nil {
+		return descruntime.SignatureInfo{}, fmt.Errorf("rsa sign: %w", err)
+	}
+
+	switch supported.GetSignatureEncodingPolicy() {
+	case v1alpha1.SignatureEncodingPolicyPlain:
+		return descruntime.SignatureInfo{
+			Algorithm: algorithm,
+			MediaType: supported.GetDefaultMediaType(),
+			Value:     hex.EncodeToString(rawSig),
+		}, nil
+
+	case v1alpha1.SignatureEncodingPolicyPEM:
+		fallthrough
+	default:
+		chain, err := rsacredentials.CertificateChainFromCredentials(creds)
+		if err != nil {
+			return descruntime.SignatureInfo{}, fmt.Errorf("read certificate chain: %w", err)
+		}
+		pem := rsasignature.SignatureBytesToPem(algorithm, rawSig, chain...)
+		return descruntime.SignatureInfo{
+			Algorithm: algorithm,
+			MediaType: v1alpha1.MediaTypePEM,
+			Value:     string(pem),
+		}, nil
+	}
+}
+
+// Verify validates an OCM signature. For plain signatures, a public key must be
+// present in credentials. For PEM signatures, the embedded chain must be valid
+// against system roots and/or the optional trust anchor in credentials.
+func (h *Handler) Verify(
+	_ context.Context,
+	signed descruntime.Signature,
+	// we use hints from the signature to determine the correct settings, so no additional config is needed
+	_ runtime.Typed,
+	creds map[string]string,
+) error {
+	pubFromCreds, underlying := rsacredentials.PublicKeyFromCredentials(creds)
+
+	hash, dig, err := parseDigest(signed.Digest)
+	if err != nil {
+		return err
+	}
+
+	switch signed.Signature.MediaType {
+	case v1alpha1.MediaTypePlainRSASSAPSS, v1alpha1.MediaTypePlainRSASSAPKCS1V15:
+		if pubFromCreds == nil {
+			return ErrMissingPublicKey
+		}
+		sig, err := hex.DecodeString(signed.Signature.Value)
+		if err != nil {
+			return fmt.Errorf("decode hex signature: %w", err)
+		}
+		alg, err := algorithmFromPlainMedia(signed.Signature.MediaType)
+		if err != nil {
+			return err
+		}
+		return verifyRSA(alg, pubFromCreds, hash, dig, sig)
+
+	case v1alpha1.MediaTypePEM:
+		sig, algFromPEM, chain, err := rsasignature.GetSignatureFromPem([]byte(signed.Signature.Value))
+		if err != nil {
+			return fmt.Errorf("parse pem signature: %w", err)
+		}
+		if len(chain) == 0 {
+			return errors.New("pem signature missing certificate chain")
+		}
+		leaf := chain[0]
+		rsaPub, ok := leaf.PublicKey.(*rsa.PublicKey)
+		if !ok {
+			return errors.New("leaf cert public key is not RSA")
+		}
+		if err := verifyChainWithOptionalAnchor(leaf, chain[1:], underlying, h.roots, h.now); err != nil {
```

Comment:
The pubslice could panic

Reply:
actually, not sure what could panic, mind explaining?

```go
sig, algFromPEM, chain, err := rsasignature.GetSignatureFromPem([]byte(signed.Signature.Value))
if err != nil {
    return fmt.Errorf("parse pem signature: %w", err)
}
if len(chain) == 0 {
    return errors.New("pem signature missing certificate chain")
}
leaf := chain[0]
...
if err := verifyChainWithOptionalAnchor(leaf, chain[1:], underlying, h.roots, h.now); err != nil {
    ...
}
``` 

- check len(chain) == 0 before accessing chain[0].
- If len(chain) == 1, then chain[1:] is a valid empty slice, not a panic.
- If len(chain) > 1, then chain[1:] is the remainder of the slice.

Reply:
`chain[1:]` I wasn't too sure about this part, but yeah after testing it you are right.